test/e2e/vpcs.e2e.ts

```diff
@@ -74,3 +74,99 @@ test('can create and delete subnet', async ({ page }) => {
+  await expect(rows).toHaveCount(1)
+  await expect(rows.getByText('mock-custom-router')).toBeHidden()
+
+  // create a new router
+  await page.click('text=New router')
+  await page.fill('input[name=name]', 'mock-custom-router-2')
+  await page.click('button:has-text("Create router")')
+  await expect(rows).toHaveCount(2)
+  await expectRowVisible(table, { name: 'mock-custom-router-2' })
+
+  // click on mock-system-router to go to the router detail page
+  await page.getByText('mock-system-router').click()
+  await expect(page).toHaveURL(
+    '/projects/mock-project/vpcs/mock-vpc/routers/mock-system-router'
+  )
+})
+
+test('can’t create or delete Routes on system routers', async ({ page }) => {
+  // load the router
+  await page.goto('/projects/mock-project/vpcs/mock-vpc/routers/mock-system-router')
+
+  // verify that the "new route" link isn't present, since users can't add routes to system routers
+  await expect(page.getByRole('link', { name: 'New route' })).toBeHidden()
+
+  // expect to see table of routes
+  const table = page.getByRole('table')
+  const routeRows = table.locator('tbody >> tr')
+  await expect(routeRows).toHaveCount(3)
+  await expectRowVisible(table, { Name: 'default' })
+  await expectRowVisible(table, { Name: 'default-v4' })
+  await expectRowVisible(table, { Name: 'default-v6' })
+  await routeRows.first().getByRole('button', { name: 'Row actions' }).click()
+  // can't delete default routes
+  await expect(page.getByRole('menuitem', { name: 'Delete' })).toBeDisabled()
+})
+
+test('can create, update, and delete Route', async ({ page }) => {
+  // go to the custom-router-2 page
+  await page.goto('/projects/mock-project/vpcs/mock-vpc/routers/mock-custom-router')
+  const table = page.getByRole('table')
+  const routeRows = table.locator('tbody >> tr')
+  await expectRowVisible(table, { Name: 'drop-local' })
+
+  // create a new route
+  await page.click('text=New route')
+  await page.getByRole('textbox', { name: 'name' }).fill('new-route')
+  await page.getByRole('textbox', { name: 'Destination value' }).fill('0.0.0.0')
+
+  // we'll set the target in a second, but first verify that selecting internet gateway disables the value
+  await page.getByRole('button', { name: 'Target type' }).click()
+  await page.getByRole('option', { name: 'Internet gateway' }).click()
+  await expect(page.getByRole('textbox', { name: 'Target value' })).toBeDisabled()
+  await expect(page.getByRole('textbox', { name: 'Target value' })).toHaveValue('outbound')
+  await page.getByRole('button', { name: 'Target type' }).click()
+  await page.getByRole('option', { name: 'IP' }).click()
+  await page.getByRole('textbox', { name: 'Target value' }).fill('1.1.1.1')
+  await page.getByRole('button', { name: 'Create route' }).click()
+  await expect(routeRows).toHaveCount(2)
+  await expectRowVisible(table, { Name: 'new-route' })
+
+  // see the destination value of 0.0.0.0
+  await expectRowVisible(table, { Destination: 'ip0.0.0.0' })
```

Comment:
this and the above can be combined into one since it's one row. same with 162-165. and you might as well assert the target value too